miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/storage/FileWriter.kt

```diff
@@ -7,16 +7,11 @@ import java.io.InputStream
 
 internal class FileWriter {
 
-    suspend fun write(
-        inputStream: InputStream,
-        parentPath: String,
-        fileName: String
-    ) {
+    suspend fun write(inputStream: InputStream, path: String) {
         withContext(Dispatchers.IO) {
-            File(parentPath).mkdirs()
-            val file = File(parentPath, fileName)
+            val file = File(path)
+            file.parentFile?.mkdirs()
             inputStream.use { it.copyTo(file.outputStream()) }
```

Comment:
If `file.parentFile` is null then the copy execution will cause crash. No further execution when the file path is invalid.
The implementation might be like this
```suggestion
            if (file.parentFile != null) {
                file.parentFile.mkdirs()
                inputStream.use { it.copyTo(file.outputStream()) }
            }
```